### ISSUE TITLE
How can I retrieve options data from browser's OPTIONS request

### ISSUE THREAD

**Issue by GabLeRoux** [axios/axios#2608]
Is there a way to retrieve browser's `OPTIONS` response data when doing a `GET` request or do I absolutely need to call `axios.options()` on my own?

I plan to generate a form based on the response data sent from an API's `OPTIONS` response I can then generate a form dynamically. Right now, when doing `OPTIONS` request on my own, it works just fine, but when I do a `GET` request, there's an additional `OPTIONS` request sent.

Just asking, but it doesn't make a big difference in the end. I guess if I want to only have a single `OPTIONS` call, I'd have to follow https://github.com/axios/axios/issues/888

**Comment by chinesedfan** [comment]
> The HTTP OPTIONS method is used to describe the communication options for the target resource.

Don't retrieve data by [OPTIONS](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS). Your additional OPTIONS request is caused by CORS. See more about CORS from [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS).

**Comment by GabLeRoux** [comment]
I’m not looking for actual data from the resource, the communication description is exactly what I’m looking for. I am using django-rest-framewrok and the OPTIONS result contains the description of what can be posted, the fields, description, field types, etc. I am using this to dynamically generate a form that is used to POST or PUT to the target in a restful way as _described_ by the api.

Right now, it works flawlessly, but there are two OPTIONS calls, the first one is the one I invoke myself because I did not figure out how I can retrieve the OPTIONS response from a simple GET request using axios.

In other words, my OPTIONS usage is perfectly fine, I’m only wondering how I can retrieve the OPTIONS response automatically sent by the browser instead of manually invoking it.

At this point in my case, it’s definitely an early optimization. It’s more a matter of knowing if it’s possible to do so.

Thanks

**Comment by chinesedfan** [comment]
@GabLeRoux Sorry for misunderstanding your problem. But as far as I know, the preflight OPTIONS request is send by the browser automatically and we can't retrieve response from it. Maybe you can ask in stackoverflow. It is an interesting question and I am glad to follow your question if it is somewhere.

**Comment by chinesedfan** [comment]
> Finally, Access-Control-Max-Age gives the value in seconds for how long the response to the preflight request can be cached for without sending another preflight request. In this case, 86400 seconds is 24 hours. Note that each browser has a maximum internal value that takes precedence when the Access-Control-Max-Age is greater.

Hope it can solve your double OPTIONS problem.
